### ISSUE TITLE
X230t support (good news inside)

### ISSUE THREAD

**Issue by hackathi** [Bizzaro/x230-osx#14]
So,

I decided to turn my tablet into a hackbook. I can add "Pen Input" to the working list at 10.12.6, however it is not rotated if you rotate the screen (have to figure out why and how I can fix this). The internal WWAN Card seems to be detected but I have not tried if i can get connectivity. At least it's showing up in Networks in System Preferences, so it's fine... I guess?

Next step is to mod my BIOS and put an AzureWave into this thing so I can leave this shitty wifi dongle and have working AirPlay over wifi. Unfortunately though it seems that I cannot get bluetooth to work correctly.

Thanks for your guide, appreciate it.

**Comment by badgeek** [comment]
how do you enable the pen input?

**Comment by hackathi** [comment]
It just worked for me; but when it does not, a shut down and restart (a.k.a cold boot) solves the problem. Internally the wacom pen input device is connected via USB. Does it show up in About this Mac -> System report (or whatever it is called)?

**Comment by badgeek** [comment]
aha touch not working but pen input works just fine!

**Comment by hackathi** [comment]
Yep, that's exactly what I've figured, too. Maybe i can port the linux driver, but I do not have any expierience writing HID drivers (wether it'd be linux or mac os). Well, a project for the holidays, I guess...

**Comment by hackathi** [comment]
FileVault2 is working too. Use [this guide](https://www.insanelymac.com/forum/topic/317290-filevault-2/?hl=%2Bclover%2B%2Bencryption) with AppleKeyFeeder.efi and you're done. However I could not get it working with the initial High Sierra installation, I updated to Mojave and encrypted after that. Works like a charm.

**Comment by Jan-E** [comment]
@mistressofjellyfish Did you mod your BIOS? If so, how? On https://www.bios-mods.com/forum/Thread-REQUEST-Lenovo-Thinkpad-X230-i-G2ETxxWW-Whitelist-Removal?pid=110094#pid110094 I only can find solutions of buying a new BIOS chip, flashing that and replacing the chip.

**Comment by pcmantinker** [comment]
If you get a Pomona clip and an SPI programmer, you can dump your BIOS (both chips) and then send the dumps to a modder at bios-mods.com. I recommend getting your BIOS updated before sending it off for modding. This way you can be sure to have the latest security patches. After you receive the modded BIOS, you can flash it back. Make sure to give your modder a nice tip. Within a day of sending off my bios for modding, I was able to flash it back and remove the whitelist.

**Comment by hackathi** [comment]
@Jan-E to be honest, by now I can't remember what I did exactly. Basically, I used a Clip and a buspirate to dump the SPI flash, make sure no Battery is plugged in, the computer is connected to the charger and Wake On Lan is on and a cable connected, to provide reliable power to the chip.

Beyond that... There is the Phoenix tool that can unpack the BIOS and somewhere a russian blog post how he removed the whitelist. I just opened the right module in my favourite reverse engineering tool, looked for the string cross reference and replaced every branch to that state in the parser logic with either a nop or a unconditional jump. I vaguely remember not sticking to the method of the russian guy, my BIOS now just accepts any PCI ID.

If that sounds too complicated to you, you should probably find someone that helps you out.

**Comment by Jan-E** [comment]
@pcmantinker @mistressofjellyfish 
Thanks for your advices. My reply is a bit late because I was waiting for the arrival of a US $5 Atheros Wifi card in the hope that it would be whitelisted. Alas: https://github.com/Bizzaro/x230-osx/issues/2#issuecomment-513951403

The good news: I just updated my X230T i7 to macOS 10.14.6 without any trouble.

**Comment by Jan-E** [comment]
BTW: there seems to be a way now to remove the Wifi whitelist without a Pomona clip and an SPI programmer.

https://github.com/n4ru/1vyrain

In the mean time, my X230t was upgraded to the latest Catalina by now: macOS 10.15.4 with the supplemental update, released 5 days ago.

**Comment by pcmantinker** [comment]
This is good news! I will have to check it out! It would be nice to flash a new BIOS as mine is a year or two old now. I enjoy using my x230t as a Windows/macOS tablet.
